capnp/src/text.rs

```diff
@@ -39,12 +40,42 @@ impl crate::introspect::Introspect for Owned {
     }
 }
 
-pub type Reader<'a> = &'a str;
+#[derive(Clone, Copy, PartialEq)]
+pub struct Reader<'a> {
+    pub reader: data::Reader<'a>,
```

Comment:
What's the benefit of making this a `data::Reader` rather than a `&[u8]` directly? The latter seems simpler to me.

Reply:
It was mostly to highlight the relation between `text` being a strict subset of `data`, and given `data::Reader` is equivalent to `&[u8]`, I figured it would make sense. I'll change it to `&[u8]` though, it's indeed simpler.